packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -3418,6 +3418,14 @@ class _History extends Iterable<_RouteEntry> with ChangeNotifier {
   String toString() {
     return _value.toString();
   }
+
+  @override
+  void dispose() {
+    for (final _RouteEntry element in _value) {
```

Comment:
This is weird, this should have been empty since the _history.dispose should only be called after `_forcedDisposeAllRouteEntries` is called (in NavigatorState.dispose()). Do you know what may have hit this code path?

It should not be this class responsibility to clear out the element in value. would be better if we add an assert empty here and trace down why it is not before disposing.

Reply:
Actually, I am not confident. This change did not fix the leak. And it seems it is better to remove it. I will update.

Reply:
removed